Slim/Web/Cometd.pm

```diff
@@ -155,6 +155,19 @@ sub handler {
 	my $delayedResponse; # false if we want to call sendResponse at the end of this method
 
 	for my $obj ( @{$objs} ) {
+	    
+		# Issue #765 - For comet meta messages handshake, connect, reconnect, disconnect, subscribe, unsubscribe populate 'id' field on response to ensure compatibility with the readily available npm comet library.
+		my $msgid;
+		if ( !$msgid ) {
+			# obtain 'id' from outer object if present, otherwise default to  - Here we use different variable name '$msgid' as '$id' is defined for use within slim/subscribe channel.
+			if ( $obj->{id} ) {
+				$msgid = $obj->{id};
+			} else {
+			    #default to empty string if not available.
+				$msgid = '';
+			}
+		}
```

Comment:
`$msgid` will always be falsy right after definition. This block could be simplified to the following, couldn't it?

```perl
my $msgid = $obj->{id} || '';
```

Reply:
Never mind, I'll do this on my end. Thanks!

Reply:
Yes Michael.  that is perfect and what I'd have done if I knew how.

I'm a VisualBasic, C/C++, Java, NodeJS, Fortran, C#, .NET guy.  
I dabbled in Perl and Python early on and to get the right syntax, I Google.

While working on this I considered the following:
**_In the /slim/request section we could move the my $id variable to the top and use it instead of my $msgid.  That way there's no confusion and $id means $id._**

Please double-check my work for all the error conditions to make sure I did not screw things up.